__#Top Free Favorable User Apps from the Apple and Google Stores__

The objective of this project is to analyze datasets from the Apple and Android/Google App stores on what type of free apps attract the most users to download and use the apps. We will be analyzing the datasets from 2018. 

We want to generate a visulation of what Free apps attracts the public to the most. Then we can choose the top 2 genres that we can produce and be profitable in both app stores. The way we generate our revenue is by the in-app ads
from the Free apps.

__##Summary of Results##__

We concluded that creating a "Reference" and a "Health & Fitness" Genere apps will be the most attractive and profitable to produce. It is a great balance of having alot of users and not being saturated. We can produce an app that brings motivation quotes or favorite lines from different books or bible passages. As well as  another genre app in the Health & Fitness that supplies bodyweight workouts for users during this time.

For more details, please refer to the the full analysis below.

Let's __START__ by opening up the two data sets:

In [94]:
#Apple Store Data

from csv import reader
opened_file = open('AppleStore.csv')
read_file = reader(opened_file)
Apple_data = list(read_file)
Apple_header = Apple_data[0]
Apple_data= Apple_data[1:]

##Android/Google App Store
from csv import reader
opened_file = open('googleplaystore.csv')
read_file = reader(opened_file)
Android_data = list(read_file)
Android_header = Android_data[0]
Android_data = Android_data[1:]


Now lets explore separtely the Apple Store data and the Android Store Data using the 'explore data function' where we can slice the data and make it more easier to read.


In [95]:
def explore_data(dataset, start, end, rows_and_columns=False):
    dataset_slice = dataset[start:end]    
    for row in dataset_slice:
        print(row)
        print('\n') # adds a new (empty) line after each row

    if rows_and_columns:
        print('Number of rows:', len(dataset))
        print('Number of columns:', len(dataset[0]))
        
print(Android_header)
print('\n')
explore_data(Android_data, 0, 3, True)

['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']


['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['Coloring book moana', 'ART_AND_DESIGN', '3.9', '967', '14M', '500,000+', 'Free', '0', 'Everyone', 'Art & Design;Pretend Play', 'January 15, 2018', '2.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


Number of rows: 10841
Number of columns: 13


We see that the Android App Store has 10841 rows and 13 Columns, we have to sift through. For the useful columns for our project we can use 'App', 'Category', 'Reviews', 'Installs', 'Price', and 'Genres'. Now lets print out Apple Store Data set

In [96]:
print(Apple_header)
print('\n')
explore_data(Apple_data,0, 3, True )

['id', 'track_name', 'size_bytes', 'currency', 'price', 'rating_count_tot', 'rating_count_ver', 'user_rating', 'user_rating_ver', 'ver', 'cont_rating', 'prime_genre', 'sup_devices.num', 'ipadSc_urls.num', 'lang.num', 'vpp_lic']


['284882215', 'Facebook', '389879808', 'USD', '0.0', '2974676', '212', '3.5', '3.5', '95.0', '4+', 'Social Networking', '37', '1', '29', '1']


['389801252', 'Instagram', '113954816', 'USD', '0.0', '2161558', '1289', '4.5', '4.0', '10.23', '12+', 'Photo & Video', '37', '0', '29', '1']


['529479190', 'Clash of Clans', '116476928', 'USD', '0.0', '2130805', '579', '4.5', '4.5', '9.24.12', '9+', 'Games', '38', '5', '18', '1']


Number of rows: 7197
Number of columns: 16


Now we see the Apple Store data where there are 7197 rows and 16 columns, and we believe that the columns of 'track_name', 'currency', 'price', 'rating_count_tot', 'rating_count_ver' and 'prime_genre'.
Not all columns makes sense; if so, please see this [link](https://www.kaggle.com/ramamet4/app-store-apple-data-set-10k-apps/home) for a full description of the columns

__#Data Cleaning!#- Deleting Wrong Data__

We want to clean these two data sets for only analyzing English speaking apps and free apps as well

In [97]:
print(Android_data[10472])  # incorrect row
print('\n')
print(Android_header)  # header
print('\n')
print(Android_data[0])      # correct row

['Life Made WI-Fi Touchscreen Photo Frame', '1.9', '19', '3.0M', '1,000+', 'Free', '0', 'Everyone', '', 'February 11, 2018', '1.0.19', '4.0 and up']


['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']


['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


The row 10472 corresponds to the app Life Made WI-Fi Touchscreen Photo Frame, and we can see that the rating is 19. This is off because the maximum rating for a Google Play app is 5. We, will delete this row.

In [102]:
print(len(Android_data))
#del Android_data[10472]  # don't run this more than once
print(len(Android_data))

# we deleted the row 10472 by using del Android_data[10427] ; only can use this function once

10841
10840


__#Deletion of Duplicates in Google/Android Data set!#__

Below you will see that the Google data set has duplicates

In [131]:
for app in Android_data[1:]:
    name = app[0]
    if name == 'Slack':
        print(app)

['Slack', 'BUSINESS', '4.4', '51507', 'Varies with device', '5,000,000+', 'Free', '0', 'Everyone', 'Business', 'August 2, 2018', 'Varies with device', 'Varies with device']
['Slack', 'BUSINESS', '4.4', '51507', 'Varies with device', '5,000,000+', 'Free', '0', 'Everyone', 'Business', 'August 2, 2018', 'Varies with device', 'Varies with device']
['Slack', 'BUSINESS', '4.4', '51510', 'Varies with device', '5,000,000+', 'Free', '0', 'Everyone', 'Business', 'August 2, 2018', 'Varies with device', 'Varies with device']


As you can see in the example above Slack has 3 duplicates in the Google/Android App data set. See below how much duplicates in total is in the Google/Android Data Set

In [104]:
duplicate_apps = []
unique_apps = []

for row in Android_data[1:]:
    name = row[0]
    if name in unique_apps:
        duplicate_apps.append(name)
    else:
        unique_apps.append(name)
    
print('Number of duplicate apps:', len(duplicate_apps))
print('\n')
print('Examples of duplicate apps: ', duplicate_apps[:15])
    

Number of duplicate apps: 1181


Examples of duplicate apps: ['Quick PDF Scanner + OCR FREE', 'Box', 'Google My Business', 'ZOOM Cloud Meetings', 'join.me - Simple Meetings', 'Box', 'Zenefits', 'Google Ads', 'Google My Business', 'Slack', 'FreshBooks Classic', 'Insightly CRM', 'QuickBooks Accounting: Invoicing & Expenses', 'HipChat - Chat Built for Teams', 'Xero Accounting Software']


We don't want to count certain apps more than once when we analyze data, so we need to remove the duplicate entries and keep only one entry per app. One thing we could do is remove the duplicate rows randomly, but we could probably find a better way.

If you examine the rows we printed two cells above for the Slack app, the main difference happens on the fourth position of each row, which corresponds to the number of reviews. The different numbers show that the data was collected at different times. We can use this to build a criterion for keeping rows. We won't remove rows randomly, but rather we'll keep the rows that have the highest number of reviews because the higher the number of reviews, the more reliable the ratings.

To do that, we will:

Create a dictionary where each key is a unique app name, and the value is the highest number of reviews of that app
Use the dictionary to create a new data set, which will have only one entry per app (and we only select the apps with the highest number of reviews)

In [100]:
for row in Android_data[1:]:
    name = row[0]
    if name == 'Box':
        print(Android_header)
        print('\n')
        print(row)  

['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']


['Box', 'BUSINESS', '4.2', '159872', 'Varies with device', '10,000,000+', 'Free', '0', 'Everyone', 'Business', 'July 31, 2018', 'Varies with device', 'Varies with device']
['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']


['Box', 'BUSINESS', '4.2', '159872', 'Varies with device', '10,000,000+', 'Free', '0', 'Everyone', 'Business', 'July 31, 2018', 'Varies with device', 'Varies with device']
['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']


['Box', 'BUSINESS', '4.2', '159872', 'Varies with device', '10,000,000+', 'Free', '0', 'Everyone', 'Business', 'July 31, 2018', 'Varies with device', 'Varies with device']


In [105]:
print('Number of rows:' , len(Android_data)- 1181)   #(remember 1181 is the number of duplicate apps)

Number of rows: 9659


__##Part Two##__

Lets start by building a Dictionary

In [106]:
reviews_max ={}    #creating an empty dictionary for the highest reviews

for app in Android_data:
    name = app[0]
    n_reviews = float(app[3])
    if name in'reviews_max' and 'reviews_max[name]' < 'n_reviews':
        reviews_max[name] =n_reviews
        
    elif name not in reviews_max:
            reviews_max[name] = n_reviews

In [107]:
print('Expected length:', len(Android_data) - 1181)
print('Actual length:', len(reviews_max))

Expected length: 9659
Actual length: 9659


This is not including the header row. Now we know the length of the dictionary Google Store Data Set

Now, let's use the reviews_max dictionary to remove the duplicates. For the duplicate cases, we'll only keep the entries with the highest number of reviews. In the code cell below:

We start by creating two empty lists, Android_clean and already_added.
We loop through the Android data set, and for every iteration:
We isolate the name of the app and the number of reviews.
'0' is the index of the App column name and '3' is the index for the column 'reviews'

We add the current row (app) to the Android_clean list, and the app name (name) to the already_added list if:
The number of reviews of the current app matches the number of reviews of that app as described in the reviews_max dictionary; and
The name of the app is not already in the already_added list. We need to add this supplementary condition to account for those cases where the highest number of reviews of a duplicate app is the same for more than one entry (for example, the Box app has three entries, and the number of reviews is the same). If we just check for reviews_max[name] == n_reviews, we'll still end up with duplicate entries for some apps.

In [108]:
Android_clean = [] #This will store our new clean data set
already_added = [] #this will store app names

for app in Android_data:
    name = app[0]
    n_reviews = float(app[3])
    if n_reviews == reviews_max[name] and name not in already_added:
        Android_clean.append(app)
        already_added.append(name)
        


In [109]:
explore_data(Android_clean, 0, 3, True)

['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['Coloring book moana', 'ART_AND_DESIGN', '3.9', '967', '14M', '500,000+', 'Free', '0', 'Everyone', 'Art & Design;Pretend Play', 'January 15, 2018', '2.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


Number of rows: 9659
Number of columns: 13


__#Removing NON-ENGLISH Apps#__

##Part One##

If you explore the data sets enough, you'll notice the names of some of the apps suggest they are not directed toward an English-speaking audience. Below, we see a couple of examples from both data sets:

In [110]:
print(Apple_data[813][1])
print(Apple_data[6731][1])

print(Android_clean[4412][0])
print(Android_clean[7940][0])

爱奇艺PPS -《欢乐颂2》电视剧热播
【脱出ゲーム】絶対に最後までプレイしないで 〜謎解き＆ブロックパズル〜
Where Am I At?
MBU DX Cluster


We want to remove the non-English apps. One way to go about this is to remove each app whose name contains a symbol that is not commonly used in English text — English text usually includes letters from the English alphabet, numbers composed of digits from 0 to 9, punctuation marks (., !, ?, ;, etc.), and other symbols (+, *, /, etc.).

All these characters that are specific to English texts are encoded using the ASCII standard. Each ASCII character has a corresponding number between 0 and 127 associated with it, and we can take advantage of that to build a function that checks an app name and tells us whether it contains non-ASCII characters.

We built this function below, and we use the built-in ord() function to find out the corresponding encoding number of each character.

In [111]:
def english_app(string):
    
    for character in string:
        if ord(character) > 130:
            return False
    
    return True
        
print(english_app('Instagram'))
print(english_app('Instachat 😜'))       

True
False


The function seems to work fine, but some English app names use emojis or other symbols (™, — (em dash), – (en dash), etc.) that fall outside of the ASCII range. Because of this, we remove useful apps if we use the function in its current form.

In [112]:
print(english_app('Docs To Go™ Free Office Suite'))
print(english_app('Instachat 😜'))

print(ord('™'))
print(ord('😜'))

False
False
8482
128540


__##Part Two##__

To minimize the impact of data loss, we'll only remove an app if its name has more than three non-ASCII characters:

In [113]:
Android_english = []
Apple_english = []

for app in Android_clean:
    name = app[0]
    if english_app(name):   #calling the english_app function to get the english names only
        Android_english.append(app)
        
for app in Apple_data:
    name = app[1]
    if english_app(name):   #call the english_app function
        Apple_english.append(app)
        
# calling the "explore_data function"
        
explore_data(Android_english, 0, 2, True)
print('\n')
explore_data(Apple_english, 0, 2, True)

['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['Coloring book moana', 'ART_AND_DESIGN', '3.9', '967', '14M', '500,000+', 'Free', '0', 'Everyone', 'Art & Design;Pretend Play', 'January 15, 2018', '2.0.0', '4.0.3 and up']


Number of rows: 9117
Number of columns: 13


['284882215', 'Facebook', '389879808', 'USD', '0.0', '2974676', '212', '3.5', '3.5', '95.0', '4+', 'Social Networking', '37', '1', '29', '1']


['389801252', 'Instagram', '113954816', 'USD', '0.0', '2161558', '1289', '4.5', '4.0', '10.23', '12+', 'Photo & Video', '37', '0', '29', '1']


Number of rows: 5707
Number of columns: 16


After getting only the English apps from both the Google/Android and Apple data set we are left with Android: 9117 and Apple: 5707 Rows

__#ISOLATING THE FREE APPS#__

We are only utlizing the information on free apps and gain its revenue for in-ad apps. Now we want to isolate the Free apps from our clean data so far.

In [114]:
Android_final_list = []
Apple_final_list = []

for app in Android_english:
    price = app[7]
    if price == '0':
        Android_final_list.append(app)
        
for app in Apple_english:
    price = app[4]
    if price == '0.0':
        Apple_final_list.append(app)

print(len(Android_final_list))
print(len(Apple_final_list))

8406
2922


We're left with 8406 Google/Android apps and 2922 Apple/iOS apps, which should be enough for our analysis and a clean data set to work with.

__So far we have done the following in our code:__

1. Removed inaccurate data
2. Removed duplicate app entries
3. Removed non-English apps
4. Isolated the free apps

#__Most Common Apps by Genre__#

__##Part One##__

Our aim is to determine the kinds of apps that are likely to attract more users because our revenue is influenced by the number of people downloading and actively using our apps.

To minimize risks and overhead, Look at our validation strategy for an app idea is comprised of __three__ steps:

Build a minimal Android version of the app, and add it to Google Play.
1. If the app has a good response from users, we then develop it further.
2. If the app is profitable after six months, we also build an iOS version of the app and add it to the App Store.
3. Because our end goal is to add the app on both the App Store and Google Play, we need to find app profiles that are successful on both markets. For instance, a profile that might work well for both markets might be a productivity app that makes use of gamification.

_Our end goal is to have app(s) that are successful in both the Android and Apple markets.

Let's begin the analysis by getting a sense of the most common genres for each market. For this, we'll build a frequency table for the prime_genre column of the App Store data set, and the Genres and Category columns of the Google Play data set.

__##Part Two##__

We'll build two functions we can use to analyze the frequency tables:

One function to generate frequency tables that show percentages
Another function that we can use to display the percentages in a descending order

In [115]:
def freq_table(dataset, index):
    table = {}   ##creating an empty dictonary 
    total = 0
    
    for row in dataset:
        total += 1
        value = row[index]
        if value in table:
            table[value] += 1
        else:
            table[value] = 1
    
    table_percentages = {}
    for key in table:
        percentage = (table[key] / total) * 100
        table_percentages[key] = percentage 
    
    return table_percentages


def display_table(dataset, index):
    table = freq_table(dataset, index)
    table_display = []
    for key in table:
        key_val_as_tuple = (table[key], key)
        table_display.append(key_val_as_tuple)
        
    table_sorted = sorted(table_display, reverse = True)
    for entry in table_sorted:
        print(entry[1], ':', entry[0])

__##Part Three##__

We start by examining the frequency table for the prime_genre column of the __Apple__ Store data set.

In [116]:
print(display_table(Apple_final_list, -5))

Games : 59.171800136892536
Entertainment : 7.529089664613278
Photo & Video : 5.133470225872689
Education : 3.8329911019849416
Social Networking : 3.1143052703627654
Shopping : 2.4982888432580426
Utilities : 2.2587268993839835
Music : 2.1560574948665296
Sports : 2.0533880903490758
Health & Fitness : 1.9849418206707734
Productivity : 1.7111567419575633
Lifestyle : 1.4715947980835045
News : 1.3347022587268993
Travel : 1.1293634496919918
Finance : 1.0951403148528405
Weather : 0.8898015058179329
Food & Drink : 0.8898015058179329
Reference : 0.5133470225872689
Business : 0.5133470225872689
Book : 0.2737850787132101
Medical : 0.20533880903490762
Navigation : 0.13689253935660506
Catalogs : 0.10266940451745381
None


 _As we can see above that in the Apple store data set the __Gaming__ genre is over 59 percent highest rating in the english apps. Following __Entertainment__ genre.


We can generalize that anything that has an entertainment factor(Social Networking, photo, games, shopping) that has more content for people to choose from and hence download. But it doesn't mean that they have the most users.


In [117]:
print(display_table(Android_final_list, 1)) #Category

FAMILY : 18.379728765167737
GAME : 9.74304068522484
TOOLS : 8.553414227932429
BUSINESS : 4.710920770877944
PRODUCTIVITY : 3.97335236735665
LIFESTYLE : 3.890078515346181
FINANCE : 3.735427075898168
MEDICAL : 3.628360694741851
PERSONALIZATION : 3.3071615512729005
SPORTS : 3.259576492981204
COMMUNICATION : 3.2238876992624315
HEALTH_AND_FITNESS : 3.1287175826790388
PHOTOGRAPHY : 3.009754936949798
NEWS_AND_MAGAZINES : 2.7956221746371637
SOCIAL : 2.664763264334999
TRAVEL_AND_LOCAL : 2.3078753271472756
SHOPPING : 2.2483940042826553
BOOKS_AND_REFERENCE : 2.1889126814180346
DATING : 1.8320247442303115
VIDEO_PLAYERS : 1.7487508922198431
MAPS_AND_NAVIGATION : 1.3561741613133478
EDUCATION : 1.2729003093028788
FOOD_AND_DRINK : 1.2015227218653344
ENTERTAINMENT : 1.1063526052819415
AUTO_AND_VEHICLES : 0.939804901261004
LIBRARIES_AND_DEMO : 0.9041161075422317
HOUSE_AND_HOME : 0.820842255531763
WEATHER : 0.7970497263859148
EVENTS : 0.7137758743754461
ART_AND_DESIGN : 0.701879609802522
PARENTING : 0.654

It seems like in the Google Play store that the Category of Family and Genre of Tools is most common. Which probably means apps that are geared toward everyday use such as (calendars, lifestyle, productivity).

Versus the Apple Store has apps geared toward FUN/Entertainment genres

Based on both the Apple and Google/Android data sets, roughly this infomation reveals the most frequent app genres that people download, but not which  kind of apps has the most users.


In [118]:
print(display_table(Android_final_list, -4)) #Genres

Tools : 8.541517963359505
Entertainment : 6.09088746133714
Education : 5.389007851534618
Business : 4.710920770877944
Productivity : 3.97335236735665
Lifestyle : 3.878182250773257
Finance : 3.735427075898168
Medical : 3.628360694741851
Sports : 3.3309540804187487
Personalization : 3.3071615512729005
Communication : 3.2238876992624315
Health & Fitness : 3.1287175826790388
Action : 3.1168213181061146
Photography : 3.009754936949798
News & Magazines : 2.7956221746371637
Social : 2.664763264334999
Travel & Local : 2.3078753271472756
Shopping : 2.2483940042826553
Books & Reference : 2.1889126814180346
Simulation : 2.081846300261718
Dating : 1.8320247442303115
Arcade : 1.8320247442303115
Casual : 1.7725434213656914
Video Players & Editors : 1.736854627646919
Maps & Navigation : 1.3561741613133478
Food & Drink : 1.2015227218653344
Puzzle : 1.1301451344277897
Racing : 1.023078753271473
Role Playing : 0.939804901261004
Auto & Vehicles : 0.939804901261004
Strategy : 0.9041161075422317
Libraries 

The difference between the Genres and the Category columns is not super clear, but one thing we can notice is that the Genres column  has more categories).

We'll only work with the Category column moving forward.

Now we'd like to get an idea about the kind of apps that have most users.

__#Most Popular Apps by Genre on the App Store#__

One way to find out what genres are the most popular (have the most users) is to calculate the average number of installs for each app genre. For the Google Play data set, we can find this information in the Installs column, but for the App Store data set this information is missing. As a workaround, we'll take the total number of user ratings as a proxy, which we can find in the rating_count_tot app.

Below, we calculate the average number of user ratings per app genre on the App Store:

In [119]:
genres_Apple= freq_table(Apple_final_list, -5)

for genre in genres_Apple:
    total = 0    #This variable will store the sum of user ratings (the number of ratings, not the actual ratings) specific to each genre.
    len_genre = 0  #This variable will store the number of apps specific to each genre.
    for app in Apple_final_list:
        genre_app = app[-5]
        if genre_app == genre:
            n_ratings = float(app[5])
            total += n_ratings
            len_genre += 1
            
    avg_n_ratings = total/ len_genre
    print(genre, ':', avg_n_ratings)

Social Networking : 78567.30769230769
Photo & Video : 29249.766666666666
Games : 21560.75072296125
Music : 55396.01587301587
Reference : 89562.6
Health & Fitness : 19418.620689655174
Weather : 48275.57692307692
Travel : 34115.57575757576
Shopping : 28877.575342465752
News : 23382.17948717949
Navigation : 125037.25
Lifestyle : 17260.53488372093
Entertainment : 15006.227272727272
Food & Drink : 33333.92307692308
Sports : 25791.666666666668
Finance : 26038.6875
Education : 6103.464285714285
Productivity : 22842.22
Utilities : 11571.69696969697
Book : 16671.0
Business : 6839.6
Catalogs : 5195.0
Medical : 612.0


Based on the results from the Apple and what is happening with our World today(April 2020) I recommend our company to create and produce more Health & Fitness apps for the public. 

Which is dominated by MyFittness Pal and Nike apps.


In [120]:
for app in Apple_final_list:
    if app[-5] == 'Health & Fitness':
        print(app[1], ':', app[5]) # print name and number of ratings

Calorie Counter & Diet Tracker by MyFitnessPal : 507706
Weight Watchers : 136833
Sleep Cycle alarm clock : 104539
Fitbit : 90496
Period Tracker Lite : 53620
Nike+ Training Club - Workouts & Fitness Plans : 33969
Plant Nanny - Water Reminder with Cute Plants : 27421
Sworkit - Custom Workouts for Exercise & Fitness : 16819
Clue Period Tracker: Period & Ovulation Tracker : 13436
Headspace : 12819
Fooducate - Lose Weight, Eat Healthy,Get Motivated : 11875
Runtastic Running, Jogging and Walking Tracker : 10298
WebMD for iPad : 9142
8fit - Workouts, meal plans and personal trainer : 8730
Record by Under Armour, connects with UA HealthBox : 7754
Fitstar Personal Trainer : 7496
My Cycles Period and Ovulation Tracker : 7469
Seven - 7 Minute Workout Training Challenge : 6808
RUNNING for weight loss: workout & meal plans : 6407
Waterlogged - Daily Hydration Tracker : 5000
J&J Official 7 Minute Workout : 4861
Fitbod Workout Log: Strength Lifting Bodybuilding : 4668
Runtastic Six Pack Abs Workout &

The same pattern applies to social networking apps, where the average number is heavily influenced by a few giants like Facebook, Pinterest, Skype, etc. Same applies to music apps, where a few big players like Pandora, Spotify, and Shazam heavily influence the average number.

Our aim is to find popular genres, but Health & Fitness, social networking or music apps might seem more popular than they really are. The average number of ratings seem to be skewed by very few apps which have hundreds of thousands of user ratings, while the other apps may struggle to get past the 5,000 threshold. Also, several apps We could get a better picture by removing these extremely popular apps for each genre and then rework the averages, but we'll leave this level of detail for later.

Reference apps have over 70K user ratings on average, but it's actually the Bible and Dictionary.com which skew up the average rating:

In [122]:
for app in Apple_final_list:
    if app[-5] == 'Reference':
        print(app[1], ':', app[5]) # print name and number of ratings

Bible : 985920
Dictionary.com Dictionary & Thesaurus : 200047
Dictionary.com Dictionary & Thesaurus for iPad : 54175
Google Translate : 26786
Muslim Pro: Ramadan 2017 Prayer Times, Azan, Quran : 18418
New Furniture Mods - Pocket Wiki & Game Tools for Minecraft PC Edition : 17588
Merriam-Webster Dictionary : 16849
Night Sky : 12122
City Maps for Minecraft PE - The Best Maps for Minecraft Pocket Edition (MCPE) : 8535
GUNS MODS for Minecraft PC Edition - Mods Tools : 1497
WWDC : 762
Horror Maps for Minecraft PE - Download The Scariest Maps for Minecraft Pocket Edition (MCPE) Free : 718
VPN Express : 14
Real Bike Traffic Rider Virtual Reality Glasses : 8
Jishokun-Japanese English Dictionary & Translator : 0


A second reccommendation for an app will be the Reference genre, as you can see above that the Bible app has many users, so something along the lines as motivation quotes or poems. Or even brain games with daily words. As we know that the "Fun" Genre is highly saturated.

Now lets analyze the Google Play Store

__#Most Popular Apps by Genre on Google Play Store#__

For the Google Play market, we actually have data about the number of installs, so we should be able to get a clearer picture about genre popularity. However, the install numbers don't seem precise enough — we can see that most values are open-ended (100+, 1,000+, 5,000+, etc.):

In [124]:
display_table(Android_final_list, 5) # the Installs columns

1,000,000+ : 15.596002855103496
100,000+ : 11.586961694028075
10,000+ : 10.456816559600284
10,000,000+ : 10.32595764929812
1,000+ : 8.482036640494885
100+ : 7.090173685462766
5,000,000+ : 6.6619081608374975
500,000+ : 5.567451820128479
50,000+ : 4.722817035450869
5,000+ : 4.532476802284083
10+ : 3.545086842731383
500+ : 3.2476802284082797
50,000,000+ : 2.1889126814180346
100,000,000+ : 2.11753509398049
50+ : 1.974779919105401
5+ : 0.820842255531763
1+ : 0.5115393766357365
500,000,000+ : 0.28551034975017847
1,000,000,000+ : 0.22602902688555793
0+ : 0.04758505829169641
0 : 0.011896264572924102


One problem with this data is that is not precise. For instance, we don't know whether an app with 100,000+ installs has 100,000 installs, 200,000, or 350,000. However, we don't need very precise data for our purposes — we only want to get an idea which app genres attract the most users, and we don't need perfect precision with respect to the number of users.

We're going to leave the numbers as they are, which means that we'll consider that an app with 100,000+ installs has 100,000 installs, and an app with 1,000,000+ installs has 1,000,000 installs, and so on.

To perform computations, however, we'll need to convert each install number to float — this means that we need to remove the commas and the plus characters, otherwise the conversion will fail and raise an error. We'll do this directly in the loop below, where we also compute the average number of installs for each genre (category).

In [125]:
categories_table = freq_table(Android_final_list, 1)

for category in categories_table:
    total = 0 #This variable will store the sum of installs specific to each genre.
    len_category = 0 #This variable will store the number of apps specific to each genre.
    for app in Android_final_list:
        category_app = app[1]
        if category_app == category:            
            n_installs = app[5]
            n_installs = n_installs.replace(',', '')
            n_installs = n_installs.replace('+', '')
            total += float(n_installs)
            len_category += 1
    avg_n_installs = total / len_category
    print(category, ':', avg_n_installs)

ART_AND_DESIGN : 1852900.0
AUTO_AND_VEHICLES : 645317.2278481013
BEAUTY : 513151.88679245283
BOOKS_AND_REFERENCE : 8504745.97826087
BUSINESS : 1602958.308080808
COMICS : 880440.625
COMMUNICATION : 36106662.328413285
DATING : 764959.4610389611
EDUCATION : 3021495.3271028036
ENTERTAINMENT : 22563010.752688173
EVENTS : 232885.83333333334
FINANCE : 1348224.9426751593
FOOD_AND_DRINK : 1974937.1386138613
HEALTH_AND_FITNESS : 4263642.1749049425
HOUSE_AND_HOME : 1371193.6376811594
LIBRARIES_AND_DEMO : 674917.2368421053
LIFESTYLE : 1375297.3058103975
GAME : 15723256.825396825
FAMILY : 2581094.698381877
MEDICAL : 119279.81639344263
SOCIAL : 24441088.17857143
SHOPPING : 7307823.2010582015
PHOTOGRAPHY : 18063710.731225297
SPORTS : 3647640.208029197
TRAVEL_AND_LOCAL : 14487541.68041237
TOOLS : 10974693.051460361
PERSONALIZATION : 5027006.791366907
PRODUCTIVITY : 16972497.946107786
PARENTING : 544745.6363636364
WEATHER : 5219216.7164179105
VIDEO_PLAYERS : 25372256.598639455
NEWS_AND_MAGAZINES : 1000

Again, the main concern is that these app genres might seem more popular than they really are. Moreover, these niches seem to be dominated by a few giants who are hard to compete against.

The game genre seems pretty popular, but previously we found out this part of the market seems a bit saturated, so we'd like to come up with a different app recommendation if possible.

The books and reference genre looks fairly popular as well, with an average number of installs of 8,504,811. It's interesting to explore this in more depth, since we found this genre has some potential to work well on the App Store, and our aim is to recommend an app genre that shows potential for being profitable on both the App Store and Google Play.

Let's take a look at some of the apps from this genre and their number of installs:

In [126]:
for app in Android_final_list:
    if app[1] == 'BOOKS_AND_REFERENCE':
        print(app[0], ':', app[5])

E-Book Read - Read Book for free : 50,000+
Download free book with green book : 100,000+
Wikipedia : 10,000,000+
Amazon Kindle : 100,000,000+
Cool Reader : 10,000,000+
Dictionary - Merriam-Webster : 10,000,000+
NOOK: Read eBooks & Magazines : 10,000,000+
Free Panda Radio Music : 100,000+
Book store : 1,000,000+
FBReader: Favorite Book Reader : 10,000,000+
English Grammar Complete Handbook : 500,000+
Free Books - Spirit Fanfiction and Stories : 1,000,000+
Google Play Books : 1,000,000,000+
AlReader -any text book reader : 5,000,000+
Offline English Dictionary : 100,000+
Oxford Dictionary of English : Free : 10,000,000+
Offline: English to Tagalog Dictionary : 500,000+
Spanish English Translator : 10,000,000+
FamilySearch Tree : 1,000,000+
Cloud of Books : 1,000,000+
Recipes of Prophetic Medicine for free : 500,000+
NOOK App for NOOK Devices : 500,000+
Anonymous caller detection : 10,000+
Ebook Reader : 5,000,000+
Litnet - E-books : 100,000+
Read books online : 5,000,000+
English to Urdu

The book and reference genre includes a variety of apps: software for processing and reading ebooks, various collections of libraries, dictionaries, tutorials on programming or languages, etc. It seems there's still a small number of extremely popular apps that skew the average:

In [127]:
for app in Android_final_list:
    if app[1] == 'BOOKS_AND_REFERENCE' and (app[5] == '1,000,000,000+'
                                            or app[5] == '500,000,000+'
                                            or app[5] == '100,000,000+'):
        print(app[0], ':', app[5])

Amazon Kindle : 100,000,000+
Google Play Books : 1,000,000,000+
Bible : 100,000,000+
Audiobooks from Audible : 100,000,000+


In [128]:
for app in Android_final_list:
    if app[1] == 'BOOKS_AND_REFERENCE' and (app[5] == '1,000,000+'
                                            or app[5] == '5,000,000+'
                                            or app[5] == '10,000,000+'
                                            or app[5] == '50,000,000+'):
        print(app[0], ':', app[5])

Wikipedia : 10,000,000+
Cool Reader : 10,000,000+
Dictionary - Merriam-Webster : 10,000,000+
NOOK: Read eBooks & Magazines : 10,000,000+
Book store : 1,000,000+
FBReader: Favorite Book Reader : 10,000,000+
Free Books - Spirit Fanfiction and Stories : 1,000,000+
AlReader -any text book reader : 5,000,000+
Oxford Dictionary of English : Free : 10,000,000+
Spanish English Translator : 10,000,000+
FamilySearch Tree : 1,000,000+
Cloud of Books : 1,000,000+
Ebook Reader : 5,000,000+
Read books online : 5,000,000+
eBoox: book reader fb2 epub zip : 1,000,000+
All Maths Formulas : 1,000,000+
Ancestry : 5,000,000+
HTC Help : 10,000,000+
Moon+ Reader : 10,000,000+
English-Myanmar Dictionary : 1,000,000+
Golden Dictionary (EN-AR) : 1,000,000+
All Language Translator Free : 1,000,000+
English Dictionary - Offline : 10,000,000+
Aldiko Book Reader : 10,000,000+
Dictionary - WordWeb : 5,000,000+
50000 Free eBooks & Free AudioBooks : 5,000,000+
Al-Quran (Free) : 10,000,000+
Al Quran Indonesia : 10,000,

This niche seems to be dominated by software for processing and reading ebooks, as well as various collections of libraries and dictionaries, so it's probably not a good idea to build similar apps since there'll be some significant competition.

Again, we notice that an app in the "Reference: genre, as you can see above that the Bible and Quran apps has many users, so something along the lines as motivation quotes or poems or bible passages app. We can even take a popular book that make it into an audible or make it into daily quotes as well.

Along with an Health and Exercise app.


__#Conclusions#__

In this project, we analyzed data about the App Store and Google Play mobile apps with the goal of recommending an app profile that can be profitable for both markets.

We concluded that creating a "Reference" and a "Health & Fitness" Genere apps will be the most attractive and profitable to produce. It is a great balance of having alot of users and not being saturated. We can produce an app that brings motivation quotes or favorite lines from different books or bible passages. As well as  another genre app in the Health & Fitness that supplies bodyweight workouts for users during this time.